pkgs/applications/misc/ilspy/default.nix

```diff
@@ -0,0 +1,99 @@
+{ lib
+, stdenv
+, pkgs
+
+, fetchzip
+, unzip
+, autoPatchelfHook
+, makeWrapper
+, makeDesktopItem
+
+#, stdenv.cc.cc.lib
+, lttng-ust
+, libkrb5
+, zlib
+, fontconfig
+
+, openssl_1_1
+, xorg
+, libX11
+, libICE
+, libSM
+, icu
+}:
+stdenv.mkDerivation rec {
+  pname = "ILSpy";
+  version = "7.2-rc";
+  description = ".NET assembly browser and decompiler.";
+
+  owner = "icsharpcode";
+  repo = "AvaloniaILSpy";
+
+  src = fetchzip {
+    url = "https://github.com/${owner}/${repo}/releases/download/v${version}/Linux.x64.Release.zip";
+    sha256 = "sha256:1crf0ng4l6x70wjlz3r6qw8l166gd52ys11j7ilb4nyy3mkjxk11";
+  };
+
+  nativeBuildInputs = [
+    unzip
+    autoPatchelfHook
+    makeWrapper
+  ];
+
+  buildInputs = [
+    stdenv.cc.cc.lib
+    lttng-ust
+    libkrb5
+    zlib
+    fontconfig
+  ];
+
+  libraryPath = lib.makeLibraryPath [
+    openssl_1_1
+    libX11
+    libICE
+    libSM
+    icu
+  ];
+
+  unpackPhase = ''unzip -qq ${src}/ILSpy-linux-x64-Release.zip'';
```

Comment:
we'd *much* rather build from source than use upstream release packages if at all possible. (no idea how .net packaging works though)

Reply:
Agreed. The build system used by this project is unfortunately significantly convoluted to get running in nix build and has no existing example of use in nixpkgs. I gave up after trying out many different integration strategies, but if someone else has the experience with either end of that equation to get it to work, that would of-course be grand :)

Reply:
if you want to do a source build of a .net project maybe look at ryujinx: https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/emulators/ryujinx/default.nix

or anything using `buildDotnetModule`

Reply:
Thanks. I have built many dot net packages just fine. This project unfortunately uses a different build system.